binderhub/static/js/index.js

```diff
@@ -252,7 +253,20 @@ function setUpLog() {
     }
   };
 
+  $('#view-raw-logs').on('click', function(ev) {
+    const blob = new Blob([logMessages.join('')], { type: 'text/plain' });
+    this.href = window.URL.createObjectURL(blob);
+    // Prevent the toggle action from firing
+    ev.stopPropagation();
+  });
```

Comment:
Is this understanding correct?
1. Line 257: The blob constant contain all log messages concatenated into a string, and of a declared type of text/plain.
2. Line 258: An data based URL destination is via createObjectURL.
3. Line 258: The link named with `id=view-raw-logs` gets its `href` field automatically set to the data based URL.

I don't understand the following though:
- Line 259-260: Hmmm... This is a click event registered on a single link element, isn't it? Would that event propagate/bubble to the toggle button somehow? What confuses me is that the toggle button isn't a parent element of the `a` link that the `view-raw-logs` element is. I'm not sure how various click events can triggered registered functions in general, but my understanding of `stopPropegation` is to mark the click event as handled somehow which I guess it is and then perhaps that's a sensible practice - but unrelated to the toggle button.

Reply:
The toggle function is attached to the header, not the button:
- https://github.com/jupyterhub/binderhub/blob/938324b95becd269da56decb4aff434af5fd9eda/binderhub/static/js/index.js#L255
- https://github.com/jupyterhub/binderhub/blob/938324b95becd269da56decb4aff434af5fd9eda/binderhub/templates/index.html#L149

Reply:
Ah, thanks!